### PR TITLE
Ooey Gooey Slimegirl Buffs Vol. 1

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
Opening as a draft while I do some number tweaks tomorrow and also handle stuff on the other PR.

How's it going.
Long have slimepeople been boring as hell. I will make that no longer a thing.
I've been glancing at https://ss13polaris.com/index.php?title=Promethean and while I haven't quite been able to nail down most of this stuff because /tg/ simply doesn't have the finely tuned species code to manage it (let me know if anyone can help me with slimes cleaning stuff they touch and getting nutrition), I've managed to port some of it. Some of it.

To begin with, slimes now have all their organs in their chest where they belong. They are **unremovable,** which means you can't really upgrade them in the round unless for some reason you take that stuff in chargen (fair play,) but now decapitation isn't that much of a problem. Let me know in the comments if you'd like to see a buff to how efficient their stomach is as a trade offer. Fits in with the whole 'eats everything.' Additionally, I went ahead and gave them more interesting names. SURGERY DOES WORK JUST AS IT USED TO.

Secondly, slimes now have easily detachable limbs. Exact testing is still to-do, but they have the same easy dismember that the ligament surgery gives you. Their limbs splatter instead of falling off, so you will **have to** regenerate them. Additionally, they are now one of the only species in the game (besides zombies) that experience decapitation. Don't worry, **the hair grows back too.** 

Now let's get to the meat and potatoes. I've got bad news and good news.

The bad news is that slimes are now weak to water. I was going to put this on various drinks and juices, but I figured it would be enough for now to just do subtypes of water. Wetstacks drain blood volume and prevent damage regeneration (more on that later), enough of them will do toxin damage. Drinking water makes the problem even worse, but it does metabolize away at about the same rate a blood filter does. Do not drown.

In exchange, slimes now have the actual potent regenerative powers that you would expect a fucking slime to have. About -2 brute and burn a tick, along with -1 oxyloss. Plasma in particular will work as a panacea for their wounds much like blood works for hemophages and plasma works for plasmamen, and in high amounts of internal jelly (like, above what it takes to split) it will work as a panacea for their various organs. Let me know if I should disable it fixing their brain.

## How This Contributes To The Nova Sector Roleplay Experience
Much like my other PRs, this is to help bring these species out of regressive and restrictive modes of balance and start to make them more mechanically deep; especially slimepeople, who require xenobiology green extracts to be 'deep' to any degree.

## Proof of Testing
<details>
<summary>Screenshots/Videos</summary>
More on that later.
</details>

## Changelog
:cl:
add: Slimes now passively heal, heal better in response to plasma, but are weak to water. Don't drink that stuff!
add: Slimes now have all their organs in their chest.
add: Slimes, also, now have much more easily detachable limbs, which splatter when severed.
/:cl:

### CODE REVIEW

**Comment by Nerev4r** [comment]
to-do list is 'making their brain removable so they don't just get RR'd when gibbed'

**Comment by Elli-Skala** [comment]
do you think it would be possible to give the brain a trait allowing it to act like all other organs? (so there is a slime core thats removeable thats really doing everything)

**Comment by zeroisthebiggay** [comment]
> do you think it would be possible to give the brain a trait allowing it to act like all other organs? (so there is a slime core thats removeable thats really doing everything)

this idea fucks hard

**Comment by Thlumyn** [comment]
I'm not quite a fan of a lot of those changes; especially being weak to water. Not being able to hang out in a pool or shower gunk off is a big loss from a rp standpoint and sounds like a bit of a pain. That regeneration is also a bit too much, imo, I don't think that they really need changes there. Moving the organs and making the limbs splatter is cool, but the rest, I'm not sure. Slimes are already very interesting mechanically, with their limb regrowth and toxin damage and shapeshifting. When I play a slime here, I want to play a slime, not a polaris Promethean

**Comment by Nerev4r** [comment]
> do you think it would be possible to give the brain a trait allowing it to act like all other organs? (so there is a slime core thats removeable thats really doing everything)

Not really possible without refactoring a lot of stuff.

**Comment by Nerev4r** [comment]
> I'm not quite a fan of a lot of those changes; especially being weak to water. Not being able to hang out in a pool or shower gunk off is a big loss from a rp standpoint and sounds like a bit of a pain. That regeneration is also a bit too much, imo, I don't think that they really need changes there. Moving the organs and making the limbs splatter is cool, but the rest, I'm not sure. Slimes are already very interesting mechanically, with their limb regrowth and toxin damage and shapeshifting. When I play a slime here, I want to play a slime, not a polaris Promethean

I'll drop a design document somewhere around the next few commits.

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 1]
```diff
@@ -133,7 +133,15 @@
 		drop_organs(phantom_owner) //Psuedoparts shouldn't have organs, but just in case
 		qdel(src)
 		return
-
+	//NOVA EDIT BEGIN - These just splatter, they don't fall off.
```
Comment (nikothedude) [inline code/modules/surgery/bodyparts/dismemberment.dm:136]: i think wounds use a different proc than this

if it works in testing it works but i just thought to warn

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 2]
```diff
@@ -133,7 +133,15 @@
...
+	for(var/obj/item/bodypart/limb as anything in phantom_owner.bodyparts)
+		if(limb.limb_id == SPECIES_SLIMEPERSON)
```
Comment (nikothedude) [inline code/modules/surgery/bodyparts/dismemberment.dm:138]: why? why not just check src instead? if src is being dismembered we and we want to splat src why are we checking ALL LIMBS

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 3]
```diff
@@ -133,7 +133,15 @@
...
+			phantom_owner.visible_message(span_notice("[phantom_owner]'s [limb] splatters with an unnerving squelch!"))
+			to_chat(phantom_owner, span_warning("Your [src] splatters with an unnerving squelch!"))
+			playsound(phantom_owner, 'sound/effects/blobattack.ogg', 60, TRUE)
+			qdel(src)
+			return
```
Comment (nikothedude) [inline code/modules/surgery/bodyparts/dismemberment.dm:143]: look below, doing this causes a signal not to be sent. make sure its still sent

Comment (nikothedude) [reply code/modules/surgery/bodyparts/dismemberment.dm:143]: nvm its not sent in the two other cases so idk

**Comment by Nerev4r** [comment]
https://hackmd.io/@StrayCatgirl/slimegirlrework

Design doc here.

**Comment by Nerev4r** [comment]
image
image
Some other pics.

**Comment by MaSvedish** [comment]
Does strange reagent work with slimes still or is it just STRICTLY plasma? 
Is the core invulnerable ? [ I'm imagining several detonations being able to just straight up qdel if it counts as an object.

**Comment by Nerev4r** [comment]
image
image
image
image
image
special thanks to macha, slimes are now microchipped. or microcrisped if you're english.
slimepeople start with the signal off, they are given an ability which allows them to toggle it on or off at will.
upon death, if the signal is set to on, a GPS signal is added to the core. when/if the core is revived, the GPS signal is removed.
a player can activate the core in-hand to begin destroying the signal, which will print a message in chat and take approximately 30 seconds, after which the core's GPS signal is destroyed.

**Comment by ChickyRooster** [comment]
Ok but why are they microchipped? I do not see why they need that extra to them, nor the lore implications.

**Comment by nikothedude** [comment]
> Does strange reagent work with slimes still or is it just STRICTLY plasma? Is the core invulnerable ? [ I'm imagining several detonations being able to just straight up qdel if it counts as an object.

Override ex_act to avoid qdelling on exploding

**Comment by maaacha** [comment]
>Is the core invulnerable ? [ I'm imagining several detonations being able to just straight up qdel if it counts as an object.

ive tested it with maxcap and it survives just fine, dunno about spamming multiple explosions however unless theres some other jank somewhere the indestructible flag should leave it mostly safe.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 4]
```diff
@@ -19,6 +31,380 @@
...
+/datum/status_effect/slime_washing
+	id = "slime_washing"
+	alert_type = null
+	status_type = STATUS_EFFECT_UNIQUE
+
+/datum/status_effect/slime_washing/tick(seconds_between_ticks, seconds_per_tick)
+	if(ishuman(owner))
+		var/mob/living/carbon/human/slime = owner
+		for(var/obj/item/slime_items in slime.get_equipped_items(include_pockets = FALSE, include_accessories = TRUE))
+			slime_items.wash(CLEAN_WASH)
```
Comment (dwasint) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:303]: You might want to watch for this as it updates_clothing() on every item washed if its in a mob, may want to add a check to wash that only updates if a clean is done.

**Comment by dtfe3** [comment]
I think it helps separate slimepeople more from other races and also brings them more in line with existing slimes and their traits, strengths and weaknesses. I believe slimepeople did need some more changes after shapeshifting became a quirk available to veterans as well as shapeshifting also being available in NIFs. These changes help make them unique again and also makes them weaker in other ways too.

I personally like the idea of the water weakness. A comment earlier mentioned they disliked that it would take away rp opportunities and while I understand their perspective, it also introduces some interesting ways to play around it. Instead of bathing in an actual pool, you could go to the holodeck and bathe in a holographic sea. Or get a pool made of some other kind of liquid. I'm not a huge fan of the idea that they can make themselves immune to their one big weakness, but depending on how serious the slowdown is, I suppose it's fine if it's purely for RP and using it in a mechanical battle wouldn't help.

If you decide to move away from the "water immunity button" option, you could go for in-game options to help create water immunity, such as a mutation exclusive to slimes. Genetics isn't the most difficult thing to finish early on, so it shouldn't be that much of a problem to get this immunity as genetics can get slime people dna from a slime mutation toxin. Hell, you could give the mutation an added feature of making you immune to slipping on wet floors so that it might be worth finding without necessarily giving it to slimes.

Overall good PR. I like it and I look forward to hopefully seeing it in game.

File: code/modules/surgery/bodyparts/dismemberment.dm (hunk 1 above)
Comment (Nerev4r) [reply code/modules/surgery/bodyparts/dismemberment.dm:136]: image
oh there we go. yeah it does work.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 5]
```diff
@@ -19,6 +31,380 @@
...
+			if(!istype(bodypart, /obj/item/bodypart/chest))
+				qdel(bodypart)
+				continue
+		new_body.visible_message(span_warning("[new_body]'s torso \"forms\" from their core, yet to form the rest."))
+		to_chat(owner, span_purple("Your torso fully forms out of your core, yet to form the rest."))
+		brainmob.mind.transfer_to(new_body)
+		return TRUE
+	return FALSE
+
+/obj/item/bodypart/head/slime
```
[the remark this answers is not in the document]
Comment (Nerev4r) [reply modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:228]: It'd be head/jelly, changing that one; it's not the roundstart version so that all slime subspecies share it.

**Comment by white-snek** [comment]
Wasnt sure about the water weakness having spedread the changes after joining blind, few of the things i ate regularly contained water but even then it didnt seem as harsh as other servers that can cripple with just 5u... environmental doesnt seem too bad either, I approve

Love the rest of the changes, always wanted to be a goo roomba.. any chance it could be extended to liquids? having their contents injected if toggled on?

Having a physcial core is a dream for lore reasons... sadly i can see it affecting those who specifically didnt have one in flavour.
Perhaps being able to have a different core colour from mutant colour 1? I dont know if 3 is used for anything..

Regeneration is amazing to have.. i do see them in the same light but always thought there would be pushback if slimes had abilities anywhere near classic lores (no organs, no bones, no wounds to fix, physical attacks not really doing much...) They would be frankly overpowered.. seems like a good middleground.

Guessing the ability to detatch limbs at will is also too much?.. just the idea of cuffing a slime is funny.
Basically frail is nice but biased from running that already.


Love the changes so far, hope they stick around.

**Comment by Nerev4r** [comment]
some minor code improvements; slime cores no longer randomly stay behind in the ghost cafe or other dusting scenarios, your whole body is dusted now. also, slime cores now have the same throw speed and range as a baseball. you know what to do.

**Comment by TetrakoHateThis** [comment]
Odd issue that transpired:
I ate a warm donk pocket as a slime, forgot this meant omnizine so obviously took some toxin damage. After going to med and drinking up plasma, the toxins were removed but the 'health' was stuck. Blood levels were fine, replenishing past 100% didn't change it. Attempting to remove excess blood didn't change it either.

**Comment by white-snek** [comment]
> Odd issue that transpired: I ate a warm donk pocket as a slime, forgot this meant omnizine so obviously took some toxin damage. After going to med and drinking up plasma, the toxins were removed but the 'health' was stuck. Blood levels were fine, replenishing past 100% didn't change it. Attempting to remove excess blood didn't change it either.

Are you using Taur parts in creator? sounds like the same issues I had... sometimes it would clear randomly. Forgot to test toxin damage with taur disabled. see #2154

**Comment by Jinshee** [comment]
Hey, this PR really makes playing a slime really fun, thanks! There's one thing I want to give feedback on, and that's the regeneration. I was in deathmatch with a slimeperson and found that I could entirely ignore the passive damage from wounds and bloodloss just by relying purely on regen that constantly is active. I think this would really benefit from a balance change along the lines of having jelly(blood) volume directly affect the rate of regeneration, so that bleeding wounds are still something for a slimeperson to be concerned about. Also, I'd personally remove the oxyloss healing as it naturally ties into the issue mentioned, as well as not making much sense for an innate regenerative ability. As it stands, once more people start figuring out that slimepeople can shrug off a lot of wounds, they're going to become a meta pick for combat roles like security.

**Comment by white-snek** [comment]
Slimes have never been affected by bloodloss. Its due to the way their blood was set up in the first place, they have TRAIT_NOBLOOD. Would require a work around to tie their own blood code to it... Alternativly bloodloss wounds could count towards wetstacks or act like hydrophobia mode, just stopping regen untill removed.

Regen also already stops below 100% but its also unlikely to happen due to how blood regeneration works.
Slimes regenerate blood at 6 times the rate of a normal organic when falling below 100% but it has a nutrition cost. (1.25 nutrition to give 1.5cl?) You would have to be starving or have blood chunked. (or be running blood deficiency)
Chopping off a limb removes.. 40-60cl? so only 25-40s are required to gain enough blood with no other influence, doubled if they instantly regrow.

Tips for fighting a slime would be aiming for limbs, necromorph style (built in frail), or spraying them with water to stop the regen.

The blood deficiency workaround may also affect things if any passive blood drain is added / around now. Being within a narrow threshold (99-100%) stops nutrition drain to make the quirk actually reasonable to live with.
Third alternative I guess would be giving the blood deficiency trait untill it is treated but that would probably cause the most issues... codewise, not just mechanically.

**Comment by Jinshee** [comment]
b8cf79a1577c94fa9f169280ace21d87

**Comment by Nerev4r** [comment]
special thanks to golden 'goldenalpharex' alpharex
limb dropping should now be more consistent less shitcode and also it now actually drains the proper amount of blood

**Comment by Krypandenej** [comment]
Trying to use a 100u beaker of plasma on a core just DNRs them

**Comment by lilBlovy** [comment]
I think having a buff where you dont get a reduced eating speed when you are overweight would be nice for the slime peeps since they would mostly disolve stuff than actually force it down somewhere

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 6]
```diff
@@ -42,7 +42,12 @@
 		BODY_ZONE_CHEST = /obj/item/bodypart/chest/jelly,
 	)
 	var/datum/action/innate/regenerate_limbs/regenerate_limbs
-	var/datum/action/innate/alter_form/alter_form //NOVA EDIT ADDITION - CUSTOMIZATION
+	// NOVA EDIT ADDITION BEGIN
+	var/datum/action/innate/alter_form/alter_form
+	var/datum/action/cooldown/spell/slime_washing/slime_washing
+	var/datum/action/cooldown/spell/slime_hydrophobia/slime_hydrophobia
+	var/datum/action/innate/core_signal/core_signal
+	//NOVA EDIT ADDITION END
```
Comment (GoldenAlpharex) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm:50]: All of these should be modularized, and documented.

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 7]
```diff
@@ -52,6 +57,12 @@
 		//NOVA EDIT ADDITION BEGIN - CUSTOMIZATION
 		alter_form = new
 		alter_form.Grant(new_jellyperson)
+		slime_washing = new
+		slime_washing.Grant(new_jellyperson)
+		slime_hydrophobia = new
+		slime_hydrophobia.Grant(new_jellyperson)
+		core_signal = new
+		core_signal.Grant(new_jellyperson)
 		//NOVA EDIT ADDITION END
```
Comment (GoldenAlpharex) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm:66]: All of this could also be modularized.

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 8]
```diff
@@ -62,6 +73,12 @@
 	//NOVA EDIT ADDITION BEGIN - CUSTOMIZATION
 	if(alter_form)
 		alter_form.Remove(former_jellyperson)
+	if(slime_washing)
+		slime_washing.Remove(former_jellyperson)
+	if(slime_hydrophobia)
+		slime_hydrophobia.Remove(former_jellyperson)
+	if(core_signal)
+		core_signal.Remove(former_jellyperson)
 	//NOVA EDIT ADDITION END
```
Comment (GoldenAlpharex) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm:82]: All of this could also be modularized.

File: modular_nova/modules/bodyparts/code/roundstartslime_bodyparts.dm [hunk 9]
```diff
@@ -25,3 +28,62 @@
...
 	icon_greyscale = BODYPART_ICON_ROUNDSTARTSLIME
 	biological_state = (BIO_FLESH|BIO_BLOODED)
+
+/obj/item/bodypart/head/jelly/drop_limb(special, dismembered, move_to_floor = FALSE)
+	if(special)
+		return ..()
+
+	to_chat(owner, span_warning("Your [name] splatters with an unnerving squelch!"))
+	playsound(owner, 'sound/effects/blobattack.ogg', 60, TRUE)
+	owner.blood_volume -= 60
```
Comment (GoldenAlpharex) [inline modular_nova/modules/bodyparts/code/roundstartslime_bodyparts.dm:38]: This value should be a define, something like `SLIME_LIMB_BLOOD_VALUE` or something like that.

File: modular_nova/modules/bodyparts/code/roundstartslime_bodyparts.dm [hunk 10]
```diff
@@ -25,3 +28,62 @@
...
+	qdel(src)
+	return .
+
+/obj/item/bodypart/arm/left/jelly/drop_limb(special, dismembered, move_to_floor = FALSE)
+	if(special)
+		return ..()
+
+	to_chat(owner, span_warning("Your [name] splatters with an unnerving squelch!"))
+	playsound(owner, 'sound/effects/blobattack.ogg', 60, TRUE)
+	owner.blood_volume -= 60
```
Comment (GoldenAlpharex) [inline modular_nova/modules/bodyparts/code/roundstartslime_bodyparts.dm:49]: (Same goes for all of the following ones here too)

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 11]
```diff
@@ -18,6 +32,382 @@
...
+	colorize()
+	core_ejected = FALSE
+	RegisterSignal(organ_owner, COMSIG_LIVING_DEATH, PROC_REF(on_slime_death))
+
+/obj/item/organ/internal/brain/slime/proc/colorize()
+	if(owner && isjellyperson(owner))
+		core_color = owner.dna.features["mcolor"]
+		add_atom_colour(core_color, FIXED_COLOUR_PRIORITY)
+
+/obj/item/organ/internal/brain/slime/proc/on_slime_death(mob/living/victim, gibbed)
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:132]: These two procs need documentation.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 12]
```diff
@@ -18,6 +32,382 @@
...
+		qdel(src)
+		UnregisterSignal(victim, COMSIG_LIVING_DEATH)
+		return
+
+	addtimer(CALLBACK(src, PROC_REF(core_ejection), victim), 0) // explode them after the current proc chain ends, to avoid weirdness
+
+///////
+/// CORE EJECTION PROC
+/// Makes it so that when a slime dies, their core ejects and their body is qdel'd.
+
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:146]: This is a better format for multiline documentation comments.
```suggestion
/**
 * CORE EJECTION PROC - 
 * Makes it so that when a slime dies, their core ejects and their body is qdel'd.
 */
```

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 13]
```diff
@@ -18,6 +32,382 @@
...
+	do_steam_effects(get_turf(victim))
+	playsound(victim, 'sound/effects/blobattack.ogg', 80, TRUE)
+
+	if(gps_active) // adding the gps signal if they have activated the ability
+		AddComponent(/datum/component/gps, "[victim]'s Core")
+
+	qdel(victim)
+	UnregisterSignal(victim, COMSIG_LIVING_DEATH)
+
+/obj/item/organ/internal/brain/slime/proc/do_steam_effects(turf/loc)
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:170]: Document this please.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 14]
```diff
@@ -18,6 +32,382 @@
...
+	var/datum/effect_system/steam_spread/steam = new()
+	steam.set_up(10, FALSE, loc)
+	steam.start()
+
+///////
+/// CHECK FOR REPAIR SECTION
+/// Makes it so that when a slime's core has plasma poured on it, it builds a new body and moves the brain into it.
+
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:178]: Format it like I just indicated, please!

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 15]
```diff
@@ -18,6 +32,382 @@
...
+		to_chat(owner, span_purple("Your torso fully forms out of your core, yet to form the rest."))
+		brainmob.mind.transfer_to(new_body)
+		return TRUE
+	return FALSE
+
+//////
+/// HEALING SECTION
+/// Handles passive healing and water damage.
+
+/datum/species/jelly/spec_life(mob/living/carbon/human/slime, seconds_per_tick, times_fired)
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:238]: Ditto.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 16]
```diff
@@ -18,6 +32,382 @@
...
+			return
+		if(slime.stat != CONSCIOUS)
+			return
+		slime.heal_overall_damage(brute = 1.5 * seconds_per_tick, burn = 1.5 * seconds_per_tick, required_bodytype = BODYTYPE_ORGANIC)
+		slime.adjustOxyLoss(-1 * seconds_per_tick)
+
+///////
+/// SLIME CLEANING ABILITY
+/// Makes it so slimes clean themselves.
+
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:270]: Ditto.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 17]
```diff
@@ -18,6 +32,382 @@
...
+	. = ..()
+
+	if(user.has_status_effect(/datum/status_effect/slime_washing))
+		slime_washing_deactivate(user)
+		return
+
+	user.apply_status_effect(/datum/status_effect/slime_washing)
+	user.visible_message(span_purple("[user]'s outer membrane starts to develop a roiling film on the outside, absorbing grime into [user.p_their()] inner layer!"), span_purple("Your outer membrane develops a roiling film on the outside, absorbing grime off yourself and your clothes; as well as the floor beneath you."))
+
+/datum/action/cooldown/spell/slime_washing/proc/slime_washing_deactivate(mob/living/carbon/human/user) //Called when you activate it again after casting the ability-- turning them off, so to say.
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:290]: Document this please.

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 18]
```diff
@@ -18,6 +32,382 @@
...
+			if (SPT_PROB(5, seconds_per_tick))
+				slime.adjust_nutrition((rand(5,25)))
+
+/datum/status_effect/slime_washing/get_examine_text()
+	return span_notice("[owner.p_Their()] outer layer is pulling in grime, filth sinking inside of their body and vanishing.")
+
+///////
+/// HYDROPHOBIA SPELL
+/// Makes it so that slimes are waterproof, but slower, and they don't regenerate.
+
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:324]: Format it like I just indicated, please!

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 19]
```diff
@@ -18,6 +32,382 @@
...
+	if(user.has_status_effect(/datum/status_effect/slime_hydrophobia))
+		slime_hydrophobia_deactivate(user)
+		return
+
+	ADD_TRAIT(user, TRAIT_SLIME_HYDROPHOBIA, ACTION_TRAIT)
+	user.apply_status_effect(/datum/status_effect/slime_hydrophobia)
+	user.visible_message(span_purple("[user]'s outer membrane starts to ooze out an oily coating, [owner.p_their()] body becoming more viscous!"), span_purple("Your outer membrane starts to ooze out an oily coating, protecting you from water but making your body more viscous."))
+
+/datum/action/cooldown/spell/slime_hydrophobia/proc/slime_hydrophobia_deactivate(mob/living/carbon/human/user)
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:345]: Document this please!

File: modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm [hunk 20]
```diff
@@ -18,6 +32,382 @@
...
+/datum/status_effect/slime_hydrophobia/on_remove()
+	. = ..()
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/status_effect/slime_hydrophobia, update=TRUE)
+
+/datum/status_effect/slime_hydrophobia/get_examine_text()
+	return span_notice("[owner.p_They()] is oozing out an oily coating onto [owner.p_their()] outer membrane, water rolling right off.")
+
+///////
+/// CHEMICAL HANDLING
+/// Here's where slimes heal off plasma and where they hate drinking water.
```
Comment (GoldenAlpharex) [inline modular_nova/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm:374]: If this isn't meant to be documentation, just make is use two slashes instead of three. In this case, I'd recon it's not meant to be proc documentation, because it hides away the proper proc documentation that's already defined in more details on `/datum/species`

File: modular_nova/modules/organs/code/stomach.dm [hunk 21]
```diff
@@ -14,6 +14,18 @@
 	maxHealth = 1.5 * STANDARD_ORGAN_THRESHOLD
 	metabolism_efficiency = 0.07
 
+/obj/item/organ/internal/stomach/slime/oversized
+	name = "huge golgi apparatus"
+	desc = "Typically found in huge slimes, this monstrous organelle has been developed to be highly efficient, made to provide an enormous amount of nutrients to an enormous ooze."
+	maxHealth = 1.5 * STANDARD_ORGAN_THRESHOLD
+	metabolism_efficiency = 0.07
+
+//Not a stomach, but suitable for where we keep oversized schtuff.
```
Comment (GoldenAlpharex) [inline modular_nova/modules/organs/code/stomach.dm:23]:
```suggestion
// Not a stomach, but suitable for where we keep oversized schtuff.
```

File: modular_nova/modules/oversized/code/oversized_quirk.dm [hunk 22]
```diff
@@ -36,7 +36,20 @@
 		return
 	old_stomach.Remove(human_holder, special = TRUE)
 	qdel(old_stomach)
-	if(issynthetic(human_holder))
+	if(isjellyperson(human_holder))
```
[the remark this answers is not in the document]
Comment (GoldenAlpharex) [reply modular_nova/modules/oversized/code/oversized_quirk.dm:34]: I'll second this, it would make this look a lot cleaner.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Sprite look great! Just a few code-related nitpicks.

**Comment by RatFromTheJungle** [comment]
doing some extensive testing with this, slimes still have no real reaction to any bleeding wounds(they don't drain blood at all), and heal fast enough to cripwalk through space with just a firesuit for no downside, which means they're essentially pod people+, that heal faster, and aren't really affected by wounds, which leaves them like, unusually above other races when it comes to survivability.

see roundstart pod people healing: 
https://github.com/NovaSector/NovaSector/blob/e3a6337c5cb3a78efe3ed64e4d105a72230bb795/modular_nova/modules/customization/modules/mob/living/carbon/human/species/podweak.dm#L44

**Comment by Nerev4r** [comment]
will take their noblood off later 👍

**Comment by white-snek** [comment]
If slimes werent heavily burdened by nutrition requirements in doing so I would have suggested making blood deficiency built in... I Kinda retract my last comment about balance.. As someone running exactly that, regeneration is rarely active unless im chugging multiple poisons or plasma... the drain is enough to keep me constantly at 99% even with just welding fuel.
Even with that I still get along fine. Constant regen isnt really needed.. Nice to have but could be more conditional/weaker

An alternative to deficiency I guess would be raising the threshold to 101%+ and putting a heavy drain if above. Slime splitting should still be doable altho will take more resources to do so..

Letting them bleed may not work as intended... altho will end up being more punishing than regular bloodloss if that is the plan, burdening nutrition first, same as if running deficiency. 
However.. poking a hole in an evershifting mass and needing to bandage/cauterise it does sound kinda weird... are there limits to how much bleeding wounds would remove? or could a slime bleed out if they cant/wont treat it?

**Comment by MaSvedish** [comment]
hey chat this isnt good. cores currently arent invulnerable as claimed. if an explosion with any sort of secondary [no devestation] occurs, it will 100% just qdel the core. its happened a few times in the last 4-5 days ive played.

**Comment by Nerev4r** [comment]
fixed

**Comment by Krypandenej** [comment]
I'm unsure if this directly related to the PR, but if you put digi legs on your slime char for whatever reason in the pref editor, said legs can still suffer bone breakages

**Comment by white-snek** [comment]
Slime cleaning seems to have stopped working, as of atleast round 11072.

image

image

Was fine a day or two ago.
